From e11aad9ff20deede59e89a00c62908ffb30b8fb6 Mon Sep 17 00:00:00 2001
From: Aaron Hill <aa1ronham@gmail.com>
Date: Sun, 7 Jul 2019 14:47:20 -0400
Subject: [PATCH] Update crossbeam-deque to 0.7

The only substantive change is to 'take_local_job'.
'Worker::pop` now returns a plain 'Option<T>', so there's
no need to loop anymore

diff --git a/src/iter/par_bridge.rs b/src/iter/par_bridge.rs
index b366de69..bc938e46 100644
--- a/src/iter/par_bridge.rs
+++ b/src/iter/par_bridge.rs
@@ -79,7 +79,8 @@ where
         C: UnindexedConsumer<Self::Item>,
     {
         let split_count = AtomicUsize::new(current_num_threads());
-        let (worker, stealer) = deque::fifo();
+        let worker = deque::Worker::new_fifo();
+        let stealer = worker.stealer();
         let done = AtomicBool::new(false);
         let iter = Mutex::new((self.iter, worker));
 
@@ -149,7 +150,7 @@ where
     {
         loop {
             match self.items.steal() {
-                Steal::Data(it) => {
+                Steal::Success(it) => {
                     folder = folder.consume(it);
                     if folder.full() {
                         return folder;
